internal/plugin/terminal/ui.go

```diff
@@ -127,6 +127,13 @@ func (s *uiServer) Events(stream pb.TerminalUIService_EventsServer) error {
 		switch ev := ev.Event.(type) {
 		case *pb.TerminalUI_Event_Line_:
 			s.Impl.Output(ev.Line.Msg, terminal.WithStyle(ev.Line.Style))
+			stream.Send(&pb.TerminalUI_Response{
+				Event: &pb.TerminalUI_Response_Input{
+					Input: &pb.TerminalUI_Event_InputResp{
+						Input: "complete",
+					},
+				},
+			})
```

Comment:
Tell me more about what this is doing... I'm confused why we're sending input data when we get a line...

Reply:
@evanphx this sends a response event to the client. Mistakenly I didn't include the line on the client's side to receive the event from the server, adding that in now (in the snippet below, would be adding `u.evc.Recv()` after line 368). The purpose of that is for the client to wait for acknowledgement from the server before closing the stream.

https://github.com/hashicorp/waypoint-plugin-sdk/blob/f080dde218a23839858871a2436df3de7639e01e/internal/plugin/terminal/ui.go#L349-L369

Reply:
Ok, I think my confusion here is why are we sending a new input even for every line? 99.99% of output is not going to have input, but this causes 100% of line outputs to send input messages back.